iceoryx_posh/include/iceoryx_posh/internal/roudi/service_registry.hpp

```diff
@@ -32,23 +34,44 @@ static const capro::IdString_t Wildcard{"*"};
 class ServiceRegistry
 {
   public:
-    static constexpr uint32_t MAX_INSTANCES_PER_SERVICE = 100u;
-    using InstanceSet_t = cxx::vector<capro::IdString_t, MAX_INSTANCES_PER_SERVICE>;
-    struct instance_t
+    enum class Error
     {
-        InstanceSet_t instanceSet;
+        INVALID_STATE,
+        SERVICE_DESCRIPTION_ALREADY_ADDED,
+        SERVICE_REGISTRY_FULL,
+        SERVICE_DESCRIPTION_INVALID,
```

Comment:
Today I talked with @MatthiasKillat about his `PrefixTree` implementation and he told me that is should be possible to insert empty strings as key. The `std::multimap` also supports empty strings as keys. Therefore we could have a `ServiceDescription` with empty strings and our code wouldn't break. Since there is no technical reason for this limitation, I would suggest to allow such `ServiceDescription` since it liberates us from the burden to handle this scenario.

Reply:
I agree, if we can simplify the logic we should do it. Refactored things to be able to store an invalid `ServiceDescription`. However, there are still `isValid()` checks on user-facing API level.

Reply:
Why is an empty `ServiceDescription` invalid? If there are no technical reasons, we should not artificially introduce and invalid `ServiceDescription`. After all it was not necessary until now. Please remove the `isValid` method.